### DISCUSSION TITLE
Cast a number to a float

### DISCUSSION THREAD

**Discussion by balhoff** [souffle-lang/souffle#2100] [Q&A] [answered]
Is there a built-in way to cast a `number` to a `float`? I couldn't figure it out, so I wrote user-defined functor (`.functor asDouble(x: number): float`), which seems to work:

```c++
#include <cstdint>
#include <cfloat>

extern "C" {

double asDouble(int32_t x) {
    return (double)x;
}

}
```

Is there another way?

**Comment by b-scholz** [comment, marked as answer]
Have you tried to_float(x)?

**Reply by balhoff** [reply]
No, that's exactly what I need! Somehow I completely missed it in the docs. Thank you.
